toolkit/tools/imagegen/installutils/installutils.go

```diff
@@ -951,79 +951,145 @@ func InstallGrubCfg(installRoot, rootDevice, bootUUID, bootPrefix string, encryp
 	const (
 		assetGrubcfgFile = "/installer/grub2/grub.cfg"
 		grubCfgFile      = "boot/grub2/grub.cfg"
+		assetGrubDefFile = "/installer/efi/grub/grub"
```

Comment:
Why are these golang changes needed at this time? How are the package level changes you are making to grub2 RPM destructive to the current image tooling flow that these changes would be required in this changeset?

Reply:
There are broadly two types of variables in our existing grub.cfg:
- Information specific to the packages installed on the system (mariner.cfg, systemd.cfg), $mariner_linux
- Information specific to a given image layout (PARTUUIDs, boot prefix)
For the latter information, it can technically be acquired at runtime with `blkid.` However, since we are doing this step already in the boot tools and we want this information in a central location (/etc/defaults/grub), it makes sense to inject it during the image build process.

Reply:
Ah I see. So the thinking here is to preset these variables in the original base image, such that subsequent invocations of `grub2-mkconfig` will already have these variables set in the /etc/grub configurations because these are baked in the image from the start. Certainly makes sense.

One issue I see is there will be a gap between images that start with an older base, without the preset variables, but then pull the newer grub2 tooling as part of standard update process. These definitions will not be present since nothing will populate the information specific to the image layout. Any way we can avoid this gap?

Also note: the primary use-case behind this `grub2-mkconfig` tooling is allow a user on the live system to safely update their grub.cfg, such that the next reboot will apply the user's desired config changes, rather than edit the grub.cfg directly. This `grub2-mkconfig` live update flow has to be bulletproof, even at the expense of first-boot optimizations.

Reply:
Correct: We could extend 00_header to conditionally acquire the partitions the "old fashioned way" if they are not present by default. That would get us around the old images problem. 

We can also discuss more of these edge cases in the meeting tomorrow, I want to make sure the unattended iso installer and all other paths are still working. I also don't exactly know what is in grubenv and so am uncertain if I broke it; `grub2-editenv list` shows nothing for me.

Reply:
> Correct: We could extend 00_header to conditionally acquire the partitions the "old fashioned way" if they are not present by default. That would get us around the old images problem.

Cool, I like this idea.
 
> We can also discuss more of these edge cases in the meeting tomorrow, I want to make sure the unattended iso installer and all other paths are still working. I also don't exactly know what is in grubenv and so am uncertain if I broke it; `grub2-editenv list` shows nothing for me.

We initialize our `grubenv` to have nothing in it at the start. But that user can use grub tooling to update the `grubenv` to add custom variables (for example, setting `kernelopts=` to add additional kernel cmdline args).
And in those cases where there are contents in `grubenv`, your `grub2-mkconfig` regeneration should preserve the present behavior of loading `grubenv` and not alter it.